pkgs/top-level/python-packages.nix

```diff
@@ -10776,6 +10774,8 @@ in modules // {
     };
   };
 
+  gdal = pkgs.gdal.override { pythonPackages = self; };
```

Comment:
if gdal provides bindings then that seems like a good reason to keep it here available as a package. However, we should do something with the name as it should reflect that it is now a python package.